application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,178 @@
+    private static final String OPTION_MESSAGE_PRIVATE = "message-private";
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private final Cache<Long, Instant> channelIdToLastCommandInvocation;
+    private final Predicate<String> modMailChannelNamePredicate;
+    private final Predicate<String> guildNamePatternPredicate;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config the config to use for this
+     */
+    public ModMailCommand(Config config) {
+        super(COMMAND_NAME, "Sends a message to the moderators", SlashCommandVisibility.GLOBAL);
+        getData().addOption(OptionType.STRING, MESSAGE, "Message to the moderators", true)
+            .addOption(OptionType.BOOLEAN, OPTION_MESSAGE_PRIVATE,
+                    "do you wish for your message to stay private", false);
+
+        channelIdToLastCommandInvocation = createCooldownCache();
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+        guildNamePatternPredicate =
+                Pattern.compile(config.getGuildNamePattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        MessageChannel messageChannel = event.getChannel();
+        if (isChannelOnCooldown(messageChannel)) {
+            event
+                .reply("Please wait a bit, this command can only be used once per %d %s.".formatted(
+                        getRemainingTime(messageChannel),
+                        COOLDOWN_DURATION_UNIT.toString().toLowerCase(Locale.US)))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        channelIdToLastCommandInvocation.put(messageChannel.getIdLong(), Instant.now());
+
+        Optional<TextChannel> auditLogChannel = getChannel(event);
+        if (auditLogChannel.isEmpty()) {
+            return;
+        }
+
+        List<ModAuditLogWriter.Attachment> attachments = getAttachments(event);
+
+        String user = getName(event);
+        MessageAction message = auditLogChannel.orElseThrow().sendMessageEmbeds(messageEmbed(user));
+
+        message = buildAttachment(attachments, message);
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    private int getRemainingTime(MessageChannel messageChannel) {
+        return COOLDOWN_DURATION_VALUE - getTimePassedSinceCommandCalled(messageChannel);
+    }
+
+    private String getName(SlashCommandInteractionEvent event) {
+        String user = event.getUser().getAsMention();
```

Comment:
Name of whom? Bad method name. Instead, maybe `getAuthorName`

Reply:
Done